pandas/tests/window/test_expanding.py

```diff
@@ -90,13 +90,60 @@ def test_missing_minp_zero(self):
         expected = pd.Series([np.nan])
         tm.assert_series_equal(result, expected)
 
-    @pytest.mark.parametrize("klass", [pd.Series, pd.DataFrame])
-    def test_iter_raises(self, klass):
-        # https://github.com/pandas-dev/pandas/issues/11704
-        # Iteration over a Window
-        obj = klass([1, 2, 3, 4])
-        with pytest.raises(NotImplementedError):
-            iter(obj.expanding(2))
+    @pytest.mark.parametrize(
+        "dataframe,expected,window",
+        [
+            (
+                DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]}),
+                [({"A": [1, 2, 3], "B": [4, 5, 6]}, [0, 1, 2])],
+                3,
+            ),
+            (
+                DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]}),
+                [
+                    ({"A": [1, 2], "B": [4, 5]}, [0, 1]),
+                    ({"A": [1, 2, 3], "B": [4, 5, 6]}, [0, 1, 2]),
+                ],
+                2,
+            ),
+            (
+                DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]}),
+                [
+                    ({"A": [1], "B": [4]}, [0]),
+                    ({"A": [1, 2], "B": [4, 5]}, [0, 1]),
+                    ({"A": [1, 2, 3], "B": [4, 5, 6]}, [0, 1, 2]),
+                ],
+                1,
+            ),
+            (DataFrame({"A": [1], "B": [4]}), [], 1337),
+            (DataFrame(), [({}, [])], 1337),
+        ],
+    )
+    def test_iterator_dataframe(self, dataframe, expected, window):
+        expected = [DataFrame(values, index=index) for (values, index) in expected]
```

Comment:
The parametrize signature should probably be `"dataframe,expected_data,index,window"`

Then the test body can be:

```
expected = DataFrame(expected_data, index=index)
result = dataframe.expanding(window)
tm.assert_frame_equal(actual, expected)
```

Reply:
The Index and values are different for each iteration, so, for each item part of the iterator we'd have a pair of (expected_values, expected_index).

Reply:
Got it, thanks for the explanation. Fine as is then.